src/mmw/apps/modeling/calcs.py

```diff
@@ -0,0 +1,107 @@
+                                           4326))
+        ), clipped_counties_with_area AS (
+            SELECT ST_Area(geom_clipped) /
+                   ST_Area(ST_SetSRID(ST_GeomFromText(%s),
+                                      4326)) AS clip_percent,
+                   clipped_counties.*
+            FROM clipped_counties
+        )
+        SELECT SUM(beef_ha * clip_percent) as beef_ha,
+               SUM(dairy_ha * clip_percent) as dairy_ha,
+               SUM(broiler_ha * clip_percent) as broiler_ha,
+               SUM(goat_ha * clip_percent) as goat_ha,
+               SUM(hog_ha * clip_percent) as hog_ha,
+               SUM(horse_ha * clip_percent) as horse_ha,
+               SUM(layer_ha * clip_percent) as layer_ha,
+               SUM(sheep_ha * clip_percent) as sheep_ha,
+               SUM(turkey_ha * clip_percent) as turkey_ha,
+               SUM(totalaeu * clip_percent) as totalaeu,
+               SUM(grazingaeu * clip_percent) as grazingaeu,
+               SUM(totaeu_ha * clip_percent) as totaeu_ha,
+               SUM(grzaeu_ha * clip_percent) as grzaeu_ha,
+               SUM(area * clip_percent) as area,
+               SUM(totalha * clip_percent) as totalha,
+               SUM(ag_ha * clip_percent) as ag_ha,
+               SUM(ctyareaha * clip_percent) as ctyareaha,
+               SUM(ag_acres * clip_percent) as ag_acres
+        FROM clipped_counties_with_area;
+        '''
+
+    with connection.cursor() as cursor:
+        cursor.execute(sql, [geom.wkt, geom.wkt, geom.wkt])
+
+        animal_pop = namedtuple('AnimalPopulation',
+                                [col[0] for col in cursor.description])
+
+        animal_data = {}
+
+        for k, v in animal_pop(*(cursor.fetchone()))._asdict().iteritems():
+            animal_data[k.decode('utf-8')] = v
+
+        animal_fields = {
+            u'beef_ha': u'Beef',
+            u'dairy_ha': u'Dairy',
+            u'broiler_ha': u'Broilers',
+            u'goat_ha': u'Goats',
+            u'hog_ha': u'Hogs',
+            u'horse_ha': u'Horses',
+            u'layer_ha': u'Layers',
+            u'sheep_ha': u'Sheep',
+            u'turkey_ha': u'Turkeys'
+        }
+
+        animal_aeu_ha = 0
+        for key in animal_fields.keys():
+            animal_aeu_ha = animal_aeu_ha + animal_data[key]
+
+        def coverage_for_animal(code, display_name, ag_ha=animal_aeu_ha):
+            if ag_ha == 0:
+                proportion = 0
+            else:
+                proportion = animal_data[code] / ag_ha
+
+            return {
+                u'code': code,
+                u'type': display_name,
+                u'proportion': proportion,
+                u'aeu': animal_data[code]
+            }
+
+        animal_coverage = []
+
+        for code, name in animal_fields.iteritems():
+            animal_coverage.append(coverage_for_animal(code, name))
+
+        return {
+            u'displayName': 'Livestock',
```

Comment:
I don't think we need to include the `u` prefix if our keys don't have any special unicode characters.